yorkie/src/main/kotlin/dev/yorkie/document/Document.kt

```diff
@@ -369,21 +370,25 @@ public class Document(public val key: Key, private val options: Options = Option
     /**
      * Triggers an event in this [Document].
      */
-    internal suspend fun publish(event: Event) {
-        when (event) {
-            is Others.Watched -> {
-                presences.first { event.changed.actorID in it.keys }
-            }
-
-            is Others.Unwatched -> {
-                presences.first { event.changed.actorID !in it.keys }
-            }
+    internal suspend fun publishPresenceEvent(event: Event.PresenceChange) {
+        val predicate: (Presences) -> Boolean = { presences ->
+            when (event) {
+                is MyPresence.Initialized -> event.initialized == presences
+                is MyPresence.PresenceChanged -> {
+                    val actorID = event.changed.actorID
+                    actorID !in presences || event.changed.presence == presences[actorID]
+                }
 
-            is MyPresence.Initialized -> {
-                presences.first { event.initialized == it }
+                is Others.Watched -> event.changed.actorID in presences
+                is Others.Unwatched -> event.changed.actorID !in presences
+                is Others.PresenceChanged -> {
+                    val actorID = event.changed.actorID
+                    actorID !in presences || event.changed.presence == presences[actorID]
+                }
             }
-
-            else -> {}
+        }
+        withTimeoutOrNull(3_000) {
+            presences.first(predicate)
```

Comment:
is it possible for presence events to be changed as below very fast that `publishPresenceEvent` for `publishPresenceEvent` is called after it is already unwatched?

```
Watched -> PresenceChanged -> Unwatched
```

Reply:
I'm not 100% sure that situation will never happen.
I'll fix it to ensure that the `PresenceChanged` event is ignored when it is already unwatched.